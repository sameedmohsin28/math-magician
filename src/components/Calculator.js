import { useState } from 'react';
import calculate from '../logic/calculate';
import './calculator.css';

function Calculator() {
  const [result, setResult] = useState({});
  const handleClick = (e) => {
    setResult(calculate(result, e.target.innerHTML));
  };

  return (
    <>
      <div className="parent-div">
        <div className="intro-line">
          <p>Lets do some math</p>
        </div>
        <div className="calculator-div">
          <div className="display-result">{result.next === null ? result.total : result.next}</div>

          <div className="row first-row">
            <button type="button" className="btn ac" onClick={handleClick}>A/C</button>
            <button type="button" className="btn plus-minus" onClick={handleClick}>+/-</button>
            <button type="button" className="btn percentage" onClick={handleClick}>%</button>
            <button type="button" className="btn divide" onClick={handleClick}>รท</button>
          </div>

          <div className="row second-row">
            <button type="button" className="btn seven" onClick={handleClick}>7</button>
            <button type="button" className="btn eight" onClick={handleClick}>8</button>
            <button type="button" className="btn nine" onClick={handleClick}>9</button>
            <button type="button" className="btn multiply" onClick={handleClick}>x</button>
          </div>

          <div className="row third-row">
            <button type="button" className="btn four" onClick={handleClick}>4</button>
            <button type="button" className="btn five" onClick={handleClick}>5</button>
            <button type="button" className="btn six" onClick={handleClick}>6</button>
            <button type="button" className="btn minus" onClick={handleClick}>-</button>
          </div>

          <div className="row fourth-row">
            <button type="button" className="btn one" onClick={handleClick}>1</button>
            <button type="button" className="btn two" onClick={handleClick}>2</button>
            <button type="button" className="btn three" onClick={handleClick}>3</button>
            <button type="button" className="btn plus" onClick={handleClick}>+</button>
          </div>

          <div className="row fifth-row">
            <button type="button" className="btn zero" onClick={handleClick}>0</button>
            <button type="button" className="btn dot" onClick={handleClick}>.</button>
            <button type="button" className="btn equal-sign" onClick={handleClick}>=</button>
          </div>
        </div>
      </div>
    </>
  );
}

export default Calculator;
