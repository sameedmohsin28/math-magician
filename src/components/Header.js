import styles from './Header.module.css';

const Header = () => (
  <>
    <header className={styles.header}>
      <h1 className={styles.heading}>Math Magicians</h1>
      <nav>
        <a href="/" className={styles.navLinks}>Home</a>
        <a href="/calculator" className={styles.navLinks}>Calculator</a>
        <a href="/quote" className={styles.navLinks}>Quote</a>
      </nav>
    </header>
  </>
);

export default Header;
