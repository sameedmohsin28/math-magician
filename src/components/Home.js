import styles from './Home.module.css';

const Home = () => (
  <div className={styles.parentDiv}>
    <h1 className={styles.heading}>
      Welcome to our page
    </h1>
    <div>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed dt</p>

      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusm</p>
    </div>
  </div>
);

export default Home;
