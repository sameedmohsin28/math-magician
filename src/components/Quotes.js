import { useEffect, useState } from 'react';

import styles from './Quotes.module.css';

const Quotes = () => {
  const [quote, setQuote] = useState([]);
  const [loading, setLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      try {
        const getData = await fetch('https://api.api-ninjas.com/v1/quotes', {
          method: 'GET',
          headers: { 'X-Api-Key': 'XHrTP6DOL5nJ3iiJTwrfMg==OpEmLQCgPTqaBmcs' },
          contentType: 'application/json',
        });
        const response = await getData.json();
        setQuote(response);
      } catch {
        setHasError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <div>Loading...</div>;
  if (hasError) return <div>Failed to load data from API</div>;

  return (
    <>
      {quote.map((item) => (
        <div key={item.author} className={styles.quotes}>
          <p>{item.quote}</p>
          <span>
            &quot;
            {item.author}
            &quot;
          </span>
        </div>
      ))}
    </>
  );
};

export default Quotes;
