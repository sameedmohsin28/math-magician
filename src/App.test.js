import renderer from 'react-test-renderer';

import calculate from './logic/calculate';
import operate from './logic/operate';
import Header from './components/Header';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quotes from './components/Quotes';

describe('Test calculate', () => {
  it('The sum of 2 and 3 is 5', () => {
    const obj = {
      total: '2',
      next: '3',
      operation: '+',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('5');
  });

  it('The difference 10 and 3 is 7', () => {
    const obj = {
      total: '10',
      next: '3',
      operation: '-',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('7');
  });

  it('The product of 4 and 6 is 24', () => {
    const obj = {
      total: '4',
      next: '6',
      operation: 'x',
    };
    expect(calculate(obj, '=').total).toBe('24');
  });

  it('The division of 10 and 5 is 2', () => {
    const obj = {
      total: '10',
      next: '5',
      operation: 'รท',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('2');
  });

  it('50 % 40 is 10', () => {
    const obj = {
      total: '50',
      next: '40',
      operation: '%',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('10');
  });
});

describe('Test operate', () => {
  it('3 + 5 = 8', () => {
    expect(operate(3, 5, '+')).toBe('8');
  });

  it('5 - 3 = 2', () => {
    expect(operate(5, 3, '-')).toBe('2');
  });

  it('5 x 3 = 15', () => {
    expect(operate(5, 3, 'x')).toBe('15');
  });

  it('6 รท 3 = 2', () => {
    expect(operate(6, 3, 'รท')).toBe('2');
  });

  it('6 % 5 = 1', () => {
    expect(operate(6, 5, '%')).toBe('1');
  });
});
