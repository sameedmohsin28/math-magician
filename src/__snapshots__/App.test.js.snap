// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check UI components render Calculator page 1`] = `
<div
  className="parent-div"
>
  <div
    className="intro-line"
  >
    <p>
      Lets do some math
    </p>
  </div>
  <div
    className="calculator-div"
  >
    <div
      className="display-result"
    />
    <div
      className="row first-row"
    >
      <button
        className="btn ac"
        onClick={[Function]}
        type="button"
      >
        A/C
      </button>
      <button
        className="btn plus-minus"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn percentage"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn divide"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row second-row"
    >
      <button
        className="btn seven"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn eight"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn nine"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn multiply"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row third-row"
    >
      <button
        className="btn four"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn five"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn six"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn minus"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row fourth-row"
    >
      <button
        className="btn one"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn two"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn three"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn plus"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row fifth-row"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn equal-sign"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Check UI components render Quote page 1`] = `null`;

exports[`Check UI components renders Header component 1`] = `
<header
  className="header"
>
  <h1
    className="heading"
  >
    Math Magicians
  </h1>
  <nav>
    <a
      className="navLinks"
      href="/"
    >
      Home
    </a>
    <a
      className="navLinks"
      href="/calculator"
    >
      Calculator
    </a>
    <a
      className="navLinks"
      href="/quote"
    >
      Quote
    </a>
  </nav>
</header>
`;

exports[`Check UI components renders Header page 1`] = `
<div
  className="parentDiv"
>
  <h1
    className="heading"
  >
    Welcome to our page
  </h1>
  <div>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed dt
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusm
    </p>
  </div>
</div>
`;

exports[`Check UI components renders Home page 1`] = `
<div
  className="parentDiv"
>
  <h1
    className="heading"
  >
    Welcome to our page
  </h1>
  <div>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed dt
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusm
    </p>
  </div>
</div>
`;
